components/nimbus/src/schema.rs

```diff
@@ -88,6 +88,37 @@ impl ExperimentMetadata for Experiment {
     }
 }
 
+pub fn parse_experiments(payload: &str) -> Result<Vec<Experiment>> {
+    // We first encode the response into a `serde_json::Value`
+    // to allow us to deserialize each experiment individually,
+    // omitting any malformed experiments
+    let value: Value = from_str(payload)?;
+    let data = value
+        .get("data")
+        .ok_or(NimbusError::InvalidExperimentFormat)?;
+    let mut res = Vec::new();
+    for exp in data
+        .as_array()
+        .ok_or(NimbusError::InvalidExperimentFormat)?
+    {
+        // XXX: In the future it would be nice if this lived in its own versioned crate so that
+        // the schema could be decoupled from the sdk so that it can be iterated on while the
+        // sdk depends on a particular version of the schema through the Cargo.toml.
```

Comment:
💯 Yes I've been hoping we could add rust bindings to the Nimbus Experiment schema that's defined in nimbus-shared so we could do exactly this.  Maybe this is something we can do next half.